File: src/main/java/com/project/pantau/dto/user/ProfileResponse.java
package com.project.pantau.dto.user;

import lombok.Builder;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record ProfileResponse(
        UUID id,
        String displayName,
        LocalDateTime joinedAt,
        long reportsCount,
        long resolvedCount
) {
}

File: internal/dto/user/profile_response.go
package user

import (
	"time"

	"github.com/google/uuid"
)

type ProfileResponse struct {
	ID            uuid.UUID `json:"id"`
	DisplayName   string    `json:"display_name"`
	JoinedAt      time.Time `json:"joined_at"`
	ReportsCount  int64     `json:"reports_count"`
	ResolvedCount int64     `json:"resolved_count"`
}
